template/src/ontology/Makefile.jinja2

```diff
@@ -621,47 +637,49 @@ $(COMPONENTSDIR)/%.owl: | $(COMPONENTSDIR)
-	if [ $(COMP) = true ]; then if cmp -s $(TMPDIR)/component-download-{{ component.filename }}.owl $(TMPDIR)/component-download-{{ component.filename }}.tmp.owl ; then echo "Component identical."; \
+	if cmp -s $(TMPDIR)/component-download-{{ component.filename }}.owl $(TMPDIR)/component-download-{{ component.filename }}.tmp.owl ; then echo "Component identical."; \
       else echo "Component is different, updating." &&\
 		cp $(TMPDIR)/component-download-{{ component.filename }}.owl $(TMPDIR)/component-download-{{ component.filename }}.tmp.owl &&\
-		$(ROBOT) annotate -i $(TMPDIR)/component-download-{{ component.filename }}.owl --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $@; fi; fi
+		$(ROBOT) annotate -i $(TMPDIR)/component-download-{{ component.filename }}.owl --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $@; fi
 
 .PRECIOUS: $(COMPONENTSDIR)/{{ component.filename }}
+endif # MIR=true
 {% elif component.use_template %}
 $(COMPONENTSDIR)/{{ component.filename }}:{% if component.templates is not none %}{% for template in component.templates %} $(TEMPLATEDIR)/{{ template }}{% endfor %}{% else %} $(TEMPLATEDIR)/{{ component.filename.split('.') | first }}.tsv{% endif %}
-	if [ $(COMP) = true ] ; then $(ROBOT) template {% if component.template_options is not none %}{{ component.template_options }}{% endif %} \
+	$(ROBOT) template {% if component.template_options is not none %}{{ component.template_options }}{% endif %} \
 		$(patsubst %, --template %, $^) \
-		$(ANNOTATE_CONVERT_FILE); fi
+		$(ANNOTATE_CONVERT_FILE)
 
 .PRECIOUS: $(COMPONENTSDIR)/{{ component.filename }}
 {% elif component.use_mappings %}
 $(COMPONENTSDIR)/{{ component.filename }}:{% if component.mappings is not none %}{% for mapping in component.mappings %} $(MAPPINGDIR)/{{ mapping }}{% endfor %}{% else %} $(MAPPINGDIR)/{{ component.filename.split('.') | first }}sssom.tsv{% endif %}
-	if [ $(COMP) = true ] ; then $(SSSOMPY) merge $^ --output $(TMPDIR)/{{ component.filename }}-merged.sssom.tsv &&\
+	$(SSSOMPY) merge $^ --output $(TMPDIR)/{{ component.filename }}-merged.sssom.tsv &&\
 		$(SSSOMPY) convert $(TMPDIR)/{{ component.filename }}-merged.sssom.tsv {{ component.sssom_tool_options }} --output $(TMPDIR)/{{ component.filename }}-converted.sssom.owl &&\
 		$(ROBOT) merge -i $(TMPDIR)/{{ component.filename }}-converted.sssom.owl \
-		$(ANNOTATE_CONVERT_FILE); fi
+		$(ANNOTATE_CONVERT_FILE)
 
 .PRECIOUS: $(COMPONENTSDIR)/{{ component.filename }}
 {% endif -%}
 {% endfor -%}
+endif # COMP=true
 {% endif -%}
 {% if project.import_group is defined -%}
 # ----------------------------------------
 # Mirroring upstream ontologies
 # ----------------------------------------
 
-IMP=true # Global parameter to bypass import generation
-MIR=true # Global parameter to bypass mirror generation
-IMP_LARGE=true # Global parameter to bypass handling of large imports
+# Set to false to bypass all code to create/refresh mirrors
+MIR=true
```

Comment:
Is it a problem that the MIR variable is declared after it is being used by the COMP section?

Reply:
Good catch. It is indeed – until it is declared (and unless of course `Make` is invoked with a declaration on the command line), `MIR` would not exist, so the test in the COMP section would always evaluate to false.

Better to declare all MIR, IMP, IMP_LARGE, COMP, and PAT variables all at the same place, somewhere near the top of the Makefile.